api/controllers/AuthController.js

```diff
@@ -248,6 +248,106 @@ module.exports = {
     );
     return { user: result, token };
   },
+  /*
+   * @api [post] /admintoken
+   * tags:
+   *   - auth
+   * summary: Admin-only route to generate a JSON web token (JWT)
+   * parameters:
+   *   - name: X-HID-TOTP
+   *     in: header
+   *     description: The TOTP token. Required if the user has 2FA enabled.
+   *     required: false
+   *     type: string
+   * requestBody:
+   *   description: 'User email and password'
+   *   required: true
+   *   content:
+   *     application/json:
+   *       schema:
+   *         $ref: '#/components/schemas/Auth'
+   * responses:
+   *   '200':
+   *     description: >-
+   *       The User object with the JWT contained in the `token` property.
+   *     content:
+   *       application/json:
+   *         schema:
+   *           $ref: '#/components/schemas/JWT'
+   *   '400':
+   *     description: 'Bad request. Missing email and/or password'
+   *   '401':
+   *     description: 'Wrong email and/or password'
+   *   '403':
+   *     description: 'Not an admin'
+   *   '429':
+   *     description: >-
+   *       The account was locked for 5 minutes because there were more than 5
+   *       unsuccessful login attempts within the last 5 minutes
+   * security: []
+   */
+  async authenticateAdmin(request) {
+    const result = await loginHelper(request);
+
+    // Before proceeding, check if user has admin perms. This is the main
+    // difference between @authenticate and @authenticateAdmin
+    if (!result.is_admin) {
+      throw Boom.forbidden();
+    }
```

Comment:
@orakili this is the only difference between `authenticate` and `authenticateAdmin` — the rest is a direct copy